tasks/main.yml

```diff
@@ -18,17 +18,26 @@
 
 # Workaroud for Ubuntu bug https://bugs.launchpad.net/ubuntu/+source/tzdata/+bug/1554806
 - block:
+
+  - name: Check if /etc/localtime is a link
+    stat:
+      path: /etc/localtime
+    register: st
+
   - name: Remove /etc/localtime [Ubuntu workaround]
     file:
-      dest: /etc/localhost
+      dest: /etc/localtime
       state: absent
+    when: (st.stat.lnk_source != "/usr/share/zoneinfo/{{ time_zone }}" ) and
+          (st.stat.lnk_source != "/usr/share/zoneinfo/Poland" )
 
   - name: Link timezone to localtime [Ubuntu workaround]
     file:
       src: "/usr/share/zoneinfo/{{ time_zone }}"
       dest: /etc/localtime
       state: link
-  when: ansible_distribution == "Ubuntu" and ansible_distribution_version == "16.04"
+    when: (not st.stat.exists)
```

Comment:
/etc/localtime alawys exists so this condition should be removed or moved to "block" section